### PR TITLE
Model submission for QAV500 sensor configuration 1 by Marble

### PR DESCRIPTION
**Pull request by nkoenig**
This air vehicle is based on the qav500 vehicle fielded by the CU team.

Sensor suite includes 1 forward facing realsense d435i and 2 pico flexx depth cameras https://pmdtec.com/picofamily/flexx/ these sensors publish both a depth image and point clouds. To accomplish this, they are modeled as rgbd cameras in the sdf file, but only point clouds and depth images are published. Please advise if this is possible with the depth_camera sensor type in the sdf file.

There is no light currently on the physical vehicle built by CU because it has not been necessary in the tunnel environments, but one was included here for simulation purposes

The motors for the quadrotor use the same motor model as the X4 vehicle with modifications to allow for 4 motors instead of 6. The motors on the physical vehicle are Tiger MN3510 motors http://store-en.tmotor.com/goods.php?id=337

Additional details are included in the specifications.md file

### CODE REVIEW

**Comment by acschang** [comment]
The initial assessment of the MARBLE QAV500 model is complete, however there are some issues to resolve before this model can be merged and utilized in a competitive setting:

- The OS-64 LIDAR has an update rate of 15 Hz in the model.sdf but the datasheet lists the update rate as 10 Hz or 20 Hz.  Please select one of those two frequency values.
- UAV seems to drift leftward in response to zero velocity command. Please resolve if this is not how the real UAV behaves.
- Please change the IMU noise model to match the parameters of the X1-X4 models.
- The validation data is necessary to complete the evaluation. If not provided, the model's maximum velocity will be limited to 4m/s and battery life set to 10min. When the model validation data can be provided later, please submit a PR updating the model.

Please address the issues noted above and then submit a pull request to update your submission by **5 JUNE**. The submissions are expected to be incorporated into the SubT Virtual Testbed pending a successful review.

**Comment by bfotheri** [comment]
Thanks for the comments. I apologize that we didn't see this until now. Perhaps we need to reconfigure our gits account notification settings because we didn't see this comment until we actually looked at the PR's on the osrf repo one by one. We have a number of small changes we've been working on for the past 2 months in our local branches that we would still like to submit a pull request for if possible. 

OS-64 LIDAR: The update rate has been updated accordingly. (We use the 20hz configuration)

UAV: The UAV has been tuned in our local branches / fork so reflect it's actual behavior.

IMU:  The Marble qav500 uses the Lord Microstrain 3DM-GX5-25 IMU/AHRS and we derived the IMU noise model using the datasheet of this IMU so our solutions in simulation (for SLAM and so forth) would match the ones on our hardware vehicle. We certainly can change the model to be the same as the X1-X4, it is just convenient to have our simulation vehicle specs match the one in lab.

We have some extenuating circumstances with access to our lab due to covid-19 so the 4m/s and 10 minute endurance are acceptable drop ins for now. In some previous email back and forths with osrf help they said the values wouldn't be needed because of this so I'm guessing the 4m/s and 10 minute endurance was the compromise you guys came up with. Thanks again for the flexibility during this time. 

I have a meeting later today with the team so we'll discuss these issues and submit a pull request to update our submission by tomorrow hopefully. Thanks!

**Comment by acschang** [comment]
Thank you for making the requested changes.  I adjusted the maximum linear velocity to be 4m/s and the battery life to be 10 minutes until validation data is provided.

**Comment by bfotheri** [comment]
Sounds great! Thank you very much.

File: subt_ros/launch/competition_init.launch [hunk 1]
```diff
@@ -18,10 +18,10 @@
     type="subt_ros_relay"
     name="subt_ros_relay"/>
 
-  <node
+  <!-- <node
     pkg="subt_ros"
     type="bridge_logger"
-    name="bridge_logger"/>
+    name="bridge_logger"/> -->
```
Comment (adlarkin) [inline subt_ros/launch/competition_init.launch:24]: Should the `bridge_logger` really be commented out?

**Comment by adlarkin** [comment]
Also, I have made a few updates (for example, adding in the optical frame publisher from #413), so please don't forget to run `git pull` before addressing the comments above.

File: submitted_models/marble_qav500_sensor_config_1/launch/vehicle_topics.launch [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        <remap from="$(arg sensor_prefix)/tof_top/depth_image/points" to="tof_top/depth/points"/>
+      </node>
+      <node
+        pkg="ros_ign_image"
+        type="image_bridge"
+        name="ros_ign_image_tof_top"
+        args="$(arg sensor_prefix)/tof_top/depth_image">
+        <remap from="$(arg sensor_prefix)/tof_top/depth_image" to="top/depth"/>
+      </node>
+    </group>
```
[the remark this answers is not in the document]
Comment (bfotheri) [reply submitted_models/marble_qav500_sensor_config_1/launch/vehicle_topics.launch:107]: I will change them all to top for clarity.

[the remark this answers is not in the document]
Comment (bfotheri) [reply submitted_models/marble_qav500_sensor_config_1/launch/vehicle_topics.launch:107]: Also I have submitted another PR to update model_pr_428 that has our updates to make it so the qav500 doesn't drift off as well. Our controllers work pretty well with it now. We generally don't manually drive it however. We tried some manual driving and the behavior seems improved.

[the remark this answers is not in the document]
Comment (adlarkin) [reply submitted_models/marble_qav500_sensor_config_1/launch/vehicle_topics.launch:107]: > I will change them all to top for clarity.

Thanks! Don't forget to change the bottom-facing camera topic as well. Once you have made these changes, feel free to let me know and I'll take another look.

> Also I have submitted another PR to update model_pr_428 that has our updates to make it so the qav500 doesn't drift off as well.

Can you provide me with the link to this PR?

[the remark this answers is not in the document]
Comment (bfotheri) [reply submitted_models/marble_qav500_sensor_config_1/launch/vehicle_topics.launch:107]: [https://github.com/osrf/subt/pull/467](url) Is the link to the PR

[the remark this answers is not in the document]
Comment (adlarkin) [reply submitted_models/marble_qav500_sensor_config_1/launch/vehicle_topics.launch:107]: Thanks. I just tried out the changes in the PR; the drift seems better and the topic naming is consistent now. Once you guys merge that PR into this branch, I can go ahead and mark this PR as approved.

**Comment by nkoenig** [comment]
I'm merging this PR into the cave_feature_release3 branch. This robot will be available when the feature release branch is merged into master.

**Pull request merged**